client/me/security-ssh-key/add-ssh-key-form.tsx

```diff
@@ -0,0 +1,97 @@
+import { Button, FormInputValidation } from '@automattic/components';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import i18n from 'i18n-calypso';
+import { ChangeEvent, useReducer } from 'react';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import TextareaAutosize from 'calypso/components/textarea-autosize';
+import { SSH_KEY_FORMATS } from './use-ssh-key-query';
+
+const PUBLIC_SSH_KEY_INPUT_ID = 'public_ssh_key';
+
+const initialState = {
+	touched: false,
+	isValid: false,
+	value: '',
+};
+
+const sshKeyValidation = new RegExp( `^(?:${ SSH_KEY_FORMATS.join( '|' ) })\\s.+` );
+
+type ReducerAction = { type: 'setValue'; value: string };
+
+const sshKeyFormReducer = ( state = initialState, action: ReducerAction ): typeof initialState => {
+	switch ( action.type ) {
+		case 'setValue':
+			return {
+				touched: true,
+				isValid: sshKeyValidation.test( action.value ),
+				value: action.value,
+			};
+
+		default:
+			return state;
+	}
+};
+
+interface AddSSHKeyFormProps {
+	addSSHKey( args: { name: string; key: string } ): void;
+	isAdding: boolean;
+}
+
+export const AddSSHKeyForm = ( { addSSHKey, isAdding }: AddSSHKeyFormProps ) => {
+	const [ { isValid, touched, value }, dispatch ] = useReducer( sshKeyFormReducer, initialState );
+
+	const { __ } = useI18n();
+	const formats = new Intl.ListFormat( i18n.getLocaleSlug() ?? 'en', {
+		style: 'long',
+		type: 'disjunction',
+	} ).format( SSH_KEY_FORMATS );
+
+	const showSSHKeyError = touched && ! isValid;
```

Comment:
Should we clear out the error once the user empties the form again?

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/36432/194367677-461258f2-c469-4354-adbe-b32527c84e5b.png">

Reply:
>Should we clear out the error once the user empties the form again?

No, because the field is in an invalid state as it has been touched. My idea was to follow the standard of form validation libraries.

Reply:
> I don't think so, because the field is in an invalid state as it has been touched. My idea was to follow the standard of form validation libraries. Let me know what you think about that!

Works for me 👍